### PR TITLE
Fix #11 - Onboard LED for pico_w not working

### PR DESCRIPTION
**Pull request by Kintar**
This builds on the inclusion of the cygw43_arch_none library from the 0.2 release and makes the following changes:

- hardware/rp2040/prebuild.cmake 
  - Add USING_CYW43 compiler definition (rather than PICO_W of some sort, in case other pico hardware uses this chip)
- hardware/rp2040/hardware.c
  - now initializes the cyw43 system if building for a pico_w
  - removes guard around init_onboard_led (see next point)
- hardware/rp2040/onboard_led.c
  - add conditional guards to swap out the appropriate functions for onboard LED access depending on pico variant
- cli/node_dev.c
  - remove pico_w specific guard against onboard LED

Built using both Windows 11 and MacOS Sonoma, tested on an official Raspberry Pi Pico W.

### CODE REVIEW

**Comment by Kintar** [comment]
Oh, and since I use JetBrains tools for most of my development I committed the .idea directory to ease the effort of other folks who use the same tools. Happy to nuke it and add to .gitignore if there's a reason you don't want it.

**Review by mcknly — CHANGES_REQUESTED** [review]
Looks good - just a couple minor changes:

1. Yes let's add `.idea/` to gitignore, I'd like to avoid bundling IDE settings for now (I use VScode and there's a lot of system-specific crap in there)
2. Minor boot print add - see inline comment

Looks good to merge after those changes are made.

File: hardware/rp2040/hardware_config.c [hunk 1]
```diff
@@ -33,6 +36,14 @@ void hardware_init(void) {
     // initialize the uart for cli/microshell first for status prints
     cli_uart_init();
 
+    // on pico w, we need to initialize the wifi chip
+#ifdef USING_CYW43
+    if(cyw43_arch_init()) {
+        uart_puts(UART_ID_CLI, timestamp());
+        uart_puts(UART_ID_CLI, "Failed to initialize CYW43 hardware.\r\n");
+    }
```
[the remark this answers is not in the document]
Comment (Kintar) [reply hardware/rp2040/hardware_config.c:44]: HAH! I thought I did.  Must only be on my wifi branch. :D

**Review by mcknly — APPROVED** [review]
Thanks! I am going to start using a pre-release branching strategy so that only release-up-revs go into main, so I'll merge this into [v0.3-pre](https://github.com/mcknly/breadboard-os/tree/v0.3-pre)

**Pull request merged**